lib/gameboard.rb

```diff
@@ -0,0 +1,121 @@
+class GameBoard 
+   
+  def initialize
+    @pieces = [0, 0, 0,
+               0, 0, 0,
+               0, 0, 0]
+    @players = []
+
+    (1..2).each { |i| add_player(i) }
+
+    @win = 0
+    
+  end
+
+  def add_piece(p)
+    @pieces[p.pos] = p
+  end
+
+  def add_player(player)
+  
+    if player == 1
+    puts "Please, enter first player name"
+    $name1 = gets.chomp
+    puts "Please, enter second player name"
+    $name2 = gets.chomp
+    puts "#{$name1}, which piece do you want? (X/O)" 
+
+      piece_set = false
+      while !piece_set
+        piece = gets.chomp.upcase
+        unless piece == "X" || piece == "O"
+          puts "Please enter X or O!"
+        else
+          piece_set = true
+        end
+      end
+      @players[1] = Player.new(piece == "X" ? 1 : 2)
+    elsif player == 2
+      @players[2] = Player.new(@players[1].type == 1 ? 2 : 1)
+    end
+  end
+
+  def player_move(player)
```

Comment:
gameboard should really only be concerned with displaying board correctly, perhaps this could be part of a `Game` class?